docs/language/learn-ql/writing-queries/debugging-queries.rst

```diff
@@ -0,0 +1,87 @@
+Query writing: common performance issues
+========================================
+
+This topic offers some simple tips on how to avoid common problems that can affect the performance of your queries.
+Before reading the tips below, it is worth reiterating a few important points about CodeQL and the QL language:
+
+- CodeQL `predicates <https://help.semmle.com/QL/ql-handbook/predicates.html>`__ and `classes <https://help.semmle.com/QL/ql-handbook/types.html#classes>`__ are evaluated to database `tables <https://en.wikipedia.org/wiki/Table_(database)>`__. Large predicates generate large tables with many rows, and are therefore expensive to compute.
+- The QL language is implemented using standard database operations and `relational algebra <https://en.wikipedia.org/wiki/Relational_algebra>`__ (such as join, projection, union, etc.). For further information about query languages and databases, see :doc:`About QL <../about-ql>`.
```

Comment:
```suggestion
- The QL language is implemented using standard database operations and `relational algebra <https://en.wikipedia.org/wiki/Relational_algebra>`__ (such as join, projection, and union). For further information about query languages and databases, see :doc:`About QL <../about-ql>`.
```

We don't really need "etc" after "such as".

Reply:
True!